packages/valory/connections/ledger/tests/test_contract_dispatcher.py

```diff
@@ -151,7 +151,7 @@ def getAddress(ledger_api=None, contract_address=None, _addr=None):  # type: ign
 
     message.callable = "getAddress"
     # Call a method present in the ABI but not in the contract package
-    with mock.patch("web3.contract.ContractFunction.call", return_value=0):
+    with mock.patch("web3.contract.contract.ContractFunction.call", return_value=0):
```

Comment:
API change

Reply:
I think this is a typo? `contract` twice?

Reply:
https://github.com/valory-xyz/web3.py/blob/70c6c719517ab76916671e79e94855cbae037b10/web3/contract/contract.py#L404

Reply:
https://github.com/valory-xyz/web3.py/blob/70c6c719517ab76916671e79e94855cbae037b10/web3/contract/__init__.py#L5-L6

Reply:
It does not have `ContractFunction` defined, the object we're trying to patch

Reply:
Correct, it does not have it on the hash you provided, but I thought you forked from `v6.0.0`, right? Take a look at the message above.

Reply:
@angrybayblade suggest using the public one (i.e. dropping one level of `contract`)